File: Frontend/placeful/lib/common/domain/dtos/shared_memory_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'shared_memory_dto.g.dart';

@JsonSerializable()
class SharedMemoryDto {
  final String memoryId;
  final String? friendFullName;
  final String? memoryTitle;
  final String? memoryDescription;
  final DateTime? memoryDate;
  final String? memoryImageUrl;
  final String? memoryLocationName;
  final double? memoryLocationLatitude;
  final double? memoryLocationLongitude;

  SharedMemoryDto({
    required this.memoryId,
    this.friendFullName,
    this.memoryTitle,
    this.memoryDescription,
    this.memoryDate,
    this.memoryImageUrl,
    this.memoryLocationName,
    this.memoryLocationLatitude,
    this.memoryLocationLongitude,
  });

  factory SharedMemoryDto.fromJson(Map<String, dynamic> json) =>
      _$SharedMemoryDtoFromJson(json);

  Map<String, dynamic> toJson() => _$SharedMemoryDtoToJson(this);
}

File: API/Placeful.Api/Models/DTOs/SharedMemoryDto.cs
using System.ComponentModel.DataAnnotations;
using Placeful.Api.Models.Entities;

namespace Placeful.Api.Models.DTOs;

public class SharedMemoryDto
{
    public Guid MemoryId { get; set; }
    public String? FriendFullName { get; set; }
    public string? MemoryTitle { get; set; }
    public string? MemoryDescription { get; set; }
    public DateTime? MemoryDate { get; set; }
    public string? MemoryLocationName { get; init; } 
    public double? MemoryLocationLatitude { get; init; }
    public double? MemoryLocationLongitude { get; init; }
    public string? MemoryImageUrl { get; set; }

    public SharedMemoryDto(SharedMemory memory)
    {
        MemoryId = memory.MemoryId;
        FriendFullName = memory.SharedWithUser?.FullName;
        MemoryTitle = memory.Memory?.Title;
        MemoryDescription = memory.Memory?.Description;
        MemoryDate = memory.Memory?.Date;
        MemoryLocationName = memory.Memory?.Location?.Name;
        MemoryLocationLatitude = memory.Memory?.Location?.Latitude;
        MemoryLocationLongitude = memory.Memory?.Location?.Longitude;
        MemoryImageUrl = memory.Memory?.ImageUrl;
    }
}